bindings/core/src/lib.rs

```diff
@@ -50,9 +50,31 @@ pub struct WalletOptions {
 }
 
 impl WalletOptions {
+    pub fn with_storage_path(mut self, storage_path: impl Into<Option<String>>) -> Self {
```

Comment:
Were these meant to change the visibility of fields ?

Reply:
no

Reply:
Why can't they be private if we have that ? We only use `WalletOptions` to build a `Wallet` in this module

Reply:
It's not about whether something could be private. Private fields should only be used when you cannot allow people to build your type from fields (i.e. OutputBuilders), otherwise you are limiting access for no good reason and making the API difficult to use.

Reply:
I don't think it makes the API difficult to use, you actually made it easier by adding the into setters so I don't see a good reason to keep the fields public. But as you want, I'll resolve.